docs/DeveloperGuide.md

```diff
@@ -70,7 +71,23 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture Diagram
+
+![](images/ArchitectureDiagram.png)
+
+The **Architecture Diagram** given above explains the high-level design of the program.
+
+#### Main components of the architecture
+
+`Main` consists of the `FinancialPlanner` class, and is in charge of starting up and shutting down the app.
+
+The program consists of five main components:
+
+* `Ui`: User interactions.
+* `Parser`: Parse user inputs.
+* `Command`: The command executor.
+* `Model`: Holds the data of the program in memory.
+* `Storage`: Reads data from, and writes data to the hard disk.
```

Comment:
Do you want to add our important classes such as cashflowlist?

Reply:
Thats all under model:)